From 06cc734a54a80749e7229a873ba74cfdece0f4a1 Mon Sep 17 00:00:00 2001
From: Jeffy Chen <jeffy.chen@rock-chips.com>
Date: Fri, 23 Oct 2020 17:37:04 +0800
Subject: [PATCH 68/98] pixman-renderer: Use configured format for shadow
 buffer

Honor the gbm-format from configuration for the shadow buffer instead of
hardcoding XRGB8888, preserving alpha channel when formats like ARGB8888
are specified.

Tested with:
/etc/xdg/weston/weston.ini:
[core]
gbm-format=argb8888

Run: weston --use-pixman

Signed-off-by: Jeffy Chen <jeffy.chen@rock-chips.com>
---
 libweston/pixman-renderer.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/libweston/pixman-renderer.c b/libweston/pixman-renderer.c
index 3918a579..4aad0015 100644
--- a/libweston/pixman-renderer.c
+++ b/libweston/pixman-renderer.c
@@ -1473,7 +1473,7 @@ pixman_renderer_output_create(struct weston_output *output,
 	output->renderer_state = po;
 
 	if (options->use_shadow)
-		po->shadow_format = pixel_format_get_info(DRM_FORMAT_XRGB8888);
+		po->shadow_format = options->format;
 
 	wl_list_init(&po->renderbuffer_list);
 
-- 
2.39.5

